suprb/optimizer/rule/mutation.py

```diff
@@ -160,3 +166,50 @@ def min_percentage(self, rule: Rule, random_state: RandomState):
         bounds = rule.match.bounds
         bounds[:, 0] -= random_state.uniform(0, self.sigma[0], size=bounds.shape[0])
         bounds[:, 1] += random_state.uniform(0, self.sigma[1], size=bounds.shape[0])
+
+
+class AdaptiveMutation(RuleMutation):
+    """Mutate with given operator until a peak is reached and then switch the mutation operator to Normal"""
+
+    def __init__(self, matching_type: MatchingFunction = None,
+                 sigma: Union[float, np.ndarray] = 0.1,
+                 operator: RuleMutation = HalfnormIncrease()):
+
+        super().__init__(matching_type=matching_type, sigma=sigma)
+        self.adapt_mutation = True
+        self.best_elitist_fitness = 0
+        self.number_of_worse_iterations = 0
+        self.worse_iteration_tolerance = 5
+        self.operator = operator
```

Comment:
please add a quick comment that we do this weirdness because of sklearn guidelines/checks